### PR TITLE
Feature 1796 places autocomplete

### PR DESCRIPTION
**Pull request by ianguerin**
Includes all changes from individual PRs 1,2,3 and 5 (contents of 4 included in earlier PR) that were individually reviewed.

The goal of this PR is to be tested using a more manual review, by running in various themes.

### CODE REVIEW

**Review by robyngit — CHANGES_REQUESTED** [review]
This is going to be fantastic @ianguerin, thanks for keeping it all so modular. I encountered some issues:
1. When there's no maps key configured in a repository, the ToolbarView breaks and the help tab view doesn't get rendered. The console shows the message: `Error rendering ToolbarView section TypeError: gmaps is null`
2. With a map key configured, I actually wasn't able to find any places. I only see the message: "No search results found."

**Comment by ianguerin** [comment]
> This is going to be fantastic @ianguerin, thanks for keeping it all so modular. I encountered some issues:
> 
> 1. When there's no maps key configured in a repository, the ToolbarView breaks and the help tab view doesn't get rendered. The console shows the message: `Error rendering ToolbarView section TypeError: gmaps is null`
> 2. With a map key configured, I actually wasn't able to find any places. I only see the message: "No search results found."

I've hidden the viewfinder panel if there is not Google Maps API key now. 
I'm curious to reproduce #2 in your list, as it works with my personal API key, so I wonder if there's an API not enabled on the key

**Comment by robyngit** [comment]
Tested with a properly configured API key, it works so well!! 🎉🎉🎉

File: src/js/models/geocoder/GoogleMapsAutocompleter.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      autocompleter = new gmaps.places.AutocompleteService();
+
+      /**
+       * Use the Google Maps Places API to get place predictions based off of a
+       * user input string as the user types.
+       * @param {string} input - User input to search for Google Maps places.
+       * @returns {Prediction[]} An array of places that could be the result the
+       * user is looking for. Most often this comes in five or less results.
+       */
+      async autocomplete(input) {
```
Comment (robyngit) [inline src/js/models/geocoder/GoogleMapsAutocompleter.js:25]:
```suggestion
      async autocomplete(input) {
        if (!input) return [];
```

**Review by robyngit — APPROVED** [review]

**Pull request merged**